### PR TITLE
MM-49862 - remove no longer needed webapp calls to action and pulsating dots

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
This PR removes two no longer needed calls to action: Message Actions pulsating dot and New badge on the forward button.

Also adds a condition for the custom status pulsating dot and update status post button where these two are just going to be shown once the user creation time has passed 7 days.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-49862

#### Related Pull Requests
https://github.com/mattermost/mattermost-webapp/pull/12374

#### Screenshots
Before:
de6c0c5b-bbc3-4782-9850-66d380e0a390
image-20230123-202907

After:
Screenshot 2023-03-20 at 12 51 55
Screenshot 2023-03-20 at 12 52 11

* The only change for this one is that won't show during the first seven days after user creation
Screenshot 2023-03-20 at 12 52 23

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/utils/utils.tsx [hunk 1]
```diff
@@ -90,6 +90,23 @@ const CLICKABLE_ELEMENTS = [
...
+    SECONDS = 's',
+    MINUTES = 'x',
+    HOURS = 'h',
+    DAYS = 'd',
+    FUTURE = 'f',
+    PAST = 'p'
+}
+
+export type TimeUnit = Exclude<TimeInformation, TimeInformation.FUTURE | TimeInformation.PAST>;
+export type TimeDirection = TimeInformation.FUTURE | TimeInformation.PAST;
```
Comment (julmondragon) [inline webapp/channels/src/utils/utils.tsx:109]: nit: What about defining `TimeUnit` and `TimeDirection` directly, and import them where needed?

File: webapp/channels/src/utils/utils.tsx [hunk 2]
```diff
@@ -265,6 +281,36 @@ export function getRemainingDaysFromFutureTimestamp(timestamp?: number): number
...
+    case TimeInformation.MILLISECONDS:
+        return timeline === TimeInformation.FUTURE ? timestamp + diff : timestamp - diff;
+    case TimeInformation.SECONDS:
+        return timeline === TimeInformation.FUTURE ? timestamp + (diff * MS_PER_SECOND) : timestamp + (diff * MS_PER_SECOND);
+    case TimeInformation.MINUTES:
+        return timeline === TimeInformation.FUTURE ? timestamp + (diff * MS_PER_MINUTE) : timestamp + (diff * MS_PER_MINUTE);
+    case TimeInformation.HOURS:
+        return timeline === TimeInformation.FUTURE ? timestamp + (diff * MS_PER_HOUR) : timestamp + (diff * MS_PER_HOUR);
+    case TimeInformation.DAYS:
+        return timeline === TimeInformation.FUTURE ? timestamp + (diff * MS_PER_DAY) : timestamp + (diff * MS_PER_DAY);
```
Comment (julmondragon) [inline webapp/channels/src/utils/utils.tsx:295]: Should not the falsy expression be subtracting from the timestamp (for Seconds, Minutes, Hours and Days)?

Comment (julmondragon) [reply webapp/channels/src/utils/utils.tsx:295]: nits:
- Whether it's a Future timeline or not could be calculated outside the switch (ie. `isFuture`).
- Maybe the amount to be added/subtracted can be calculated outside the ternary.
- Milliseconds seems to be the default case, could be set along with the `default`.

**Review by julmondragon — COMMENTED** [review]
Looks good, thanks @pvev !
Just some comments and nits:

**Review by nickmisasi — APPROVED** [review]
One comment but otherwise nothing to add. Thanks!

File: webapp/channels/src/utils/utils.tsx (hunk 1 above)
Comment (pvev) [reply webapp/channels/src/utils/utils.tsx:109]: I rather to keep as it is right now, TimeInformation is an enum which is intended to contain this types and maybe more if needed and TimeUnit and TimeDirection are intended to be just types.

**Review by stevemudie — APPROVED** [review]
Looks good @pvev!

**Comment by agnivade** [comment]
@pvev - Just FYI that we were testing the cherry pick functionality. We actually didn't mean to cherry-pick your commit :)

**Pull request merged**